api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -225,6 +227,12 @@ message GrpcJsonTranscoder {
   // :ref:`proto descriptor set <config_grpc_json_generate_proto_descriptor_set>`.
   bool convert_grpc_status = 9;
 
+  // Specifies a proto transformer extension that transforms grpc response status into a custom protobuf format.
```

Comment:
From a high-level I agree that there should be support for transforming the response status, however I'm not sure if we gain much be specifying this to return only a protobuf message. Why not return any valid header value?

cc @lizan that also commented on the issue (#22429).

Reply:
Sorry, I don't quite follow. We are trying to transform the grpc status format, the output of which should be written in body. Headers won't changed.